civis_jupyter_ext/magics/query.py

```diff
@@ -9,24 +11,77 @@ def magic(line, cell=None):
 
     This magic works both as a cell magic (for table previews) and a
     line magic to query a table and return a DataFrame.
+
+    You can pass either a database name or an ID in the below. If your database
+    name contains a space and you are using the line magic, you must quote
+    delimit the database.
+
+    If your version of Python supports it, F-string interpolation will work on
+    the line in both the line and cell magics but not the cell of a cell magic.
+    This isn't currently supported in IPython.
+
+    Examples
+    --------
+
+    Use the cell magic
+
+    >>> %%civisquery DATABASE
+    ... SELECT * FROM schema.table;
+
+    Use the line magic
+
+    >>> %civisquery DATABASE QUERY
+
+    Use f-strings with the line magic
+
+    >>> %civisquery {DATABASE_ID} {SQL_STATEMENT}
+
+    Use f-strings with the cell magic
+
+    >>> %%civisquery {DATABASE}
+    SELECT * FROM schema.table;
+
+    Query a database name with spaces
+
+    >>> %civisquery "My Database" {SQL_STATEMENT}
+
+    Cell magic for a database with a space
+
+    >>> %%civisquery My Database
+    SELECT * FROM schema.table
     """
 
     client = civis.APIClient()
 
     if cell is None:
-        # Not using maxsplit kwarg b/c it is not compatible w/ Python 2
-        items = [s for s in line.split(';', 1) if len(s) > 0]
-        # allow spaces
-        if len(items) == 1:
-            database, sql = items[0].split(' ', 1)
-        else:
-            database, sql = items
+        database, *lines = next(csv.reader(
+            [line],
+            delimiter=" ",
+            quotechar='"',
+            doublequote=True,
+            skipinitialspace=True)
+        )
+        sql = " ".join(lines)
+        try:
+            # if it's an integer, read_civis_sql will let it pass through
+            # if it's a string, it tries to look up a database name which
+            # will fail
```

Comment:
Will fail if the input is actually a database ID, you mean? String database names are fine.

Reply:
Clarified the language and force pushed.